app/models/developer_key.rb

```diff
@@ -101,6 +101,14 @@ def account_name
     account.try(:name)
   end
 
+  def access_token_count
+    self.access_tokens.count
+  end
+
+  def last_used_at
+    self.access_tokens.order(last_used_at: :desc).first&.last_used_at
```

Comment:
Since there's no index on developer_key_id, the performance may not be better with a last_used_at index alone (and the planner might be right in not using it), since you'll have interleaved tuples from all developer keys. I'd recommend adding a composite index on (developer_key_id, last_used_at). This (index scan) will offer better read performance than two indexes (bitmap index scan, where it has synthesize two giant bitmaps), which in turn is generally better than a sequential scan.